providers/implementations/exchange/dh_exch.c

```diff
@@ -393,6 +398,8 @@ static int dh_set_ctx_params(void *vpdhctx, const OSSL_PARAM params[])
         if (!OSSL_PARAM_get_utf8_string(p, &str, sizeof(name)))
             return 0;
         pdhctx->kdf_cekalg = OPENSSL_strdup(name);
+        if (pdhctx->kdf_cekalg == NULL)
+            return 0;
```

Comment:
I think this param should be handled in the same way as the kdf_ukm above (of course except for this is utf8 string and not an octet string.